aliasing/api/combat.py

```diff
@@ -524,19 +532,24 @@ def id(self):
         """
         return self._group.id
 
-    def get_combatant(self, name):
+    def get_combatant(self, name, strict=None):
         """
-        Gets a :class:`~aliasing.api.combat.SimpleCombatant`, fuzzy searching (partial match) on name.
+       Gets a :class:`~aliasing.api.combat.SimpleCombatant` from the group.
 
         :param str name: The name of the combatant to get.
-        :return: The combatant.
+        :param strict: Whether combatant name must be a full case insensitive match.
+                       None  - strict and then partial match. Default.
+                       False - only partial match
+                       True  - only strict match
+        :return: The combatant or None.
         :rtype: :class:`~aliasing.api.combat.SimpleCombatant`
         """
         name = str(name)
-        combatant = next((c for c in self.combatants if name.lower() in c.name.lower()), None)
-        if combatant:
-            return combatant
-        return None
+        if strict or strict is None:
+            combatant = next((c for c in self.get_combatants() if name.lower() == c.name.lower()), None)
+        if not combatant and not strict:
```

Comment:
This goes for all three places you edited: If `combatant`/`grp` doesn't get declared, this will error

Reply:
You even had `combatant = None` in your example in #contrib haha

Reply:
I totally did, not sure why I took that out. Likely during some copy-paste mangling.